src/values/ObjectValue.js

```diff
@@ -240,6 +241,9 @@ export default class ObjectValue extends ConcreteValue {
   intrinsicNameGenerated: void | true;
   hashValue: void | number;
 
+  // ReactElement
+  $BailOut: void | string;
```

Comment:
Tracked objects must be `void` or a (subtype of a) `Value`. After state joins, an `AbstractValue` may appear (encoding a conditional expression), or possibly other `Value` kinds due to simplification rules.

This is not currently enforced well, and we likely have bugs. I just opened #1124 to track this.

So... For you, that means, either...
- change type of `$BailOut` to be `void | StringValue | AbstractValue`.
- where you consume the `$BailOut`, it's probably reasonable to just give up on an `AbstractValue`.